### PR TITLE
Remove `device_vector`s from parquet

### PR DESCRIPTION
**Pull request by devavret**
Removes `device_vector` in favour of either `device_uvector` or `device_buffer` as appropriate in parquet reader and writer.

Contributes to #7287 
Depends on #7758

### CODE REVIEW

**Comment by harrism** [comment]
Any benchmark results you can show to verify no regression?

**Review by harrism — APPROVED** [review]

**Comment by vuule** [comment]
rerun tests

File: cpp/src/io/parquet/writer_impl.cu [hunk 1]
```diff
@@ -680,16 +682,17 @@ column_view parquet_column_view::leaf_column_view() const
...
 {
   column_view col  = leaf_column_view();
   auto desc        = gpu::parquet_column_device_view{};  // Zero out all fields
   desc.stats_dtype = schema_node.stats_dtype;
   desc.ts_scale    = schema_node.ts_scale;
 
-  // TODO (dm): Enable dictionary for list after refactor
-  if (physical_type() != BOOLEAN && physical_type() != UNDEFINED_TYPE && !is_list()) {
-    alloc_dictionary(_data_count);
+  // TODO (dm): Enable dictionary for list and struct after refactor
```
Comment (vuule) [inline cpp/src/io/parquet/writer_impl.cu:692]: Maybe file an issue (if you haven't already)?

**Review by vuule — APPROVED** [review]
Approved, assuming that performance did not regress.

**Review by hyperbolic2346 — CHANGES_REQUESTED** [review]
Looks good overall. Just a question here, mostly a nit.

File: cpp/include/cudf/detail/utilities/vector_factories.hpp [hunk 2]
```diff
@@ -28,6 +30,51 @@
...
+ * @return A device_uvector containing zeros
+ */
+template <typename T>
+rmm::device_uvector<T> make_zeroed_device_uvector_sync(
+  std::size_t size,
+  rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource())
+{
+  rmm::device_uvector<T> ret(size, stream, mr);
+  CUDA_TRY(cudaMemsetAsync(ret.data(), 0, size * sizeof(T), stream.value()));
```
[the remark this answers is not in the document]
Comment (devavret) [reply cpp/include/cudf/detail/utilities/vector_factories.hpp:73]: Good point. But this is from the dependency PR #7758. cc @harrism

**Comment by devavret** [comment]
> Any benchmark results you can show to verify no regression?

No noticeable effect on benchmarks.
<details><summary>Benchmark results</summary>
<p>

```
(cudf_dev) dmakkar@devavret-X99-E:/Users/dmakkar/Developer/Porting/cudf/cpp/build$ ./googlebenchmark/googlebenchmark/tools/compare.py benchmarks oldwrite newwrite Comparing oldwrite to newwrite
Benchmark                                                                          Time             CPU      Time Old      Time New       CPU Old       CPU New
---------------------------------------------------------------------------------------------------------------------------------------------------------------
ParquetWrite/integral_file_output/28/0/1/1/0/manual_time                        +0.0204         +0.0091          1056          1077           459           464
ParquetWrite/integral_file_output/28/1000/1/1/0/manual_time                     +0.0724         -0.0389           429           460           204           196
ParquetWrite/integral_file_output/28/0/32/1/0/manual_time                       -0.0269         -0.0426           128           124            89            85
ParquetWrite/integral_file_output/28/1000/32/1/0/manual_time                    -0.0110         -0.0536            89            88            67            63
ParquetWrite/integral_file_output/28/0/1/0/0/manual_time                        +0.0330         +0.0740          1038          1072           327           352
ParquetWrite/integral_file_output/28/1000/1/0/0/manual_time                     +0.0297         -0.0195           430           443           185           181
ParquetWrite/integral_file_output/28/0/32/0/0/manual_time                       +0.0735         -0.0169           111           119            70            69
ParquetWrite/integral_file_output/28/1000/32/0/0/manual_time                    -0.0128         -0.0260            86            85            62            60
ParquetWrite/integral_buffer_output/28/0/1/1/1/manual_time                      -0.0021         -0.0021           684           683           684           683
ParquetWrite/integral_buffer_output/28/1000/1/1/1/manual_time                   -0.0145         -0.0145           322           317           322           317
ParquetWrite/integral_buffer_output/28/0/32/1/1/manual_time                     -0.0181         -0.0211            93            91            91            89
ParquetWrite/integral_buffer_output/28/1000/32/1/1/manual_time                  -0.0263         -0.0277            68            67            68            66
ParquetWrite/integral_buffer_output/28/0/1/0/1/manual_time                      +0.0027         +0.0025           573           574           572           574
ParquetWrite/integral_buffer_output/28/1000/1/0/1/manual_time                   -0.0067         -0.0068           293           291           293           291
ParquetWrite/integral_buffer_output/28/0/32/0/1/manual_time                     -0.0611         -0.0550            86            80            85            80
ParquetWrite/integral_buffer_output/28/1000/32/0/1/manual_time                  -0.0301         -0.0199            68            66            67            66
ParquetWrite/integral_void_output/28/0/1/1/2/manual_time                        -0.0131         -0.0158           190           188           189           186
ParquetWrite/integral_void_output/28/1000/1/1/2/manual_time                     -0.0104         -0.0308            80            79            80            78
ParquetWrite/integral_void_output/28/0/32/1/2/manual_time                       -0.0384         -0.0386            70            67            70            67
ParquetWrite/integral_void_output/28/1000/32/1/2/manual_time                    -0.0413         -0.0478            56            53            56            53
ParquetWrite/integral_void_output/28/0/1/0/2/manual_time                        -0.0834         -0.0711            46            42            45            42
ParquetWrite/integral_void_output/28/1000/1/0/2/manual_time                     -0.0546         -0.0448            56            53            55            53
ParquetWrite/integral_void_output/28/0/32/0/2/manual_time                       -0.0349         -0.0515            50            48            50            47
ParquetWrite/integral_void_output/28/1000/32/0/2/manual_time                    -0.0429         -0.0555            51            49            50            48
ParquetWrite/floats_file_output/30/0/1/1/0/manual_time                          +0.0419         +0.0272          1021          1063           369           379
ParquetWrite/floats_file_output/30/1000/1/1/0/manual_time                       +0.0033         +0.0090           326           327           151           152
ParquetWrite/floats_file_output/30/0/32/1/0/manual_time                         +0.0045         -0.0249            97            98            59            57
ParquetWrite/floats_file_output/30/1000/32/1/0/manual_time                      -0.0372         -0.0445            60            58            45            43
ParquetWrite/floats_file_output/30/0/1/0/0/manual_time                          +0.0393         +0.0239           957           995           299           307
ParquetWrite/floats_file_output/30/1000/1/0/0/manual_time                       +0.0027         -0.0006           320           321           136           136
ParquetWrite/floats_file_output/30/0/32/0/0/manual_time                         -0.0375         -0.0144            96            93            55            54
ParquetWrite/floats_file_output/30/1000/32/0/0/manual_time                      -0.0179         -0.0409            60            59            43            42
ParquetWrite/floats_buffer_output/30/0/1/1/1/manual_time                        +0.0117         +0.0117           556           563           556           563
ParquetWrite/floats_buffer_output/30/1000/1/1/1/manual_time                     +0.0124         +0.0083           198           200           198           199
ParquetWrite/floats_buffer_output/30/0/32/1/1/manual_time                       +0.0512         +0.0490            65            68            64            68
ParquetWrite/floats_buffer_output/30/1000/32/1/1/manual_time                    -0.0350         -0.0354            46            44            46            44
ParquetWrite/floats_buffer_output/30/0/1/0/1/manual_time                        +0.0039         +0.0040           509           511           509           511
ParquetWrite/floats_buffer_output/30/1000/1/0/1/manual_time                     -0.0156         -0.0156           184           181           184           182
ParquetWrite/floats_buffer_output/30/0/32/0/1/manual_time                       -0.0333         -0.0240            68            66            68            66
ParquetWrite/floats_buffer_output/30/1000/32/0/1/manual_time                    -0.0516         -0.0527            46            44            46            43
ParquetWrite/floats_void_output/30/0/1/1/2/manual_time                          -0.0192         -0.0295            92            91            92            89
ParquetWrite/floats_void_output/30/1000/1/1/2/manual_time                       -0.0354         -0.0343            60            58            59            57
ParquetWrite/floats_void_output/30/0/32/1/2/manual_time                         -0.0740         -0.0538            40            37            39            37
ParquetWrite/floats_void_output/30/1000/32/1/2/manual_time                      -0.0669         -0.0591            37            35            37            34
ParquetWrite/floats_void_output/30/0/1/0/2/manual_time                          -0.0685         -0.0798            26            24            26            23
ParquetWrite/floats_void_output/30/1000/1/0/2/manual_time                       -0.0501         -0.0499            44            42            44            42
ParquetWrite/floats_void_output/30/0/32/0/2/manual_time                         -0.0681         -0.0599            36            34            36            33
ParquetWrite/floats_void_output/30/1000/32/0/2/manual_time                      -0.0637         -0.0580            36            34            36            34
ParquetWrite/timestamps_file_output/32/0/1/1/0/manual_time                      -0.0305         -0.0131          1097          1063           441           436
ParquetWrite/timestamps_file_output/32/1000/1/1/0/manual_time                   +0.0013         -0.0107           290           290           130           129
ParquetWrite/timestamps_file_output/32/0/32/1/0/manual_time                     -0.0401         -0.0468            96            93            60            58
ParquetWrite/timestamps_file_output/32/1000/32/1/0/manual_time                  -0.0502         -0.0484            54            52            39            38
ParquetWrite/timestamps_file_output/32/0/1/0/0/manual_time                      +0.0238         -0.0436           961           984           309           296
ParquetWrite/timestamps_file_output/32/1000/1/0/0/manual_time                   +0.0080         +0.0153           271           273           115           117
ParquetWrite/timestamps_file_output/32/0/32/0/0/manual_time                     +0.0205         -0.0367            84            86            50            48
ParquetWrite/timestamps_file_output/32/1000/32/0/0/manual_time                  -0.0087         -0.0373            52            52            38            37
ParquetWrite/timestamps_buffer_output/32/0/1/1/1/manual_time                    -0.0398         -0.0345           664           637           660           637
ParquetWrite/timestamps_buffer_output/32/1000/1/1/1/manual_time                 -0.0029         -0.0072           185           185           185           184
ParquetWrite/timestamps_buffer_output/32/0/32/1/1/manual_time                   -0.0597         -0.0577            73            68            72            68
ParquetWrite/timestamps_buffer_output/32/1000/32/1/1/manual_time                -0.0516         -0.0444            41            39            41            39
ParquetWrite/timestamps_buffer_output/32/0/1/0/1/manual_time                    +0.0312         +0.0313           557           574           557           574
ParquetWrite/timestamps_buffer_output/32/1000/1/0/1/manual_time                 -0.0086         -0.0086           170           168           170           168
ParquetWrite/timestamps_buffer_output/32/0/32/0/1/manual_time                   +0.0273         +0.0281            59            61            59            61
ParquetWrite/timestamps_buffer_output/32/1000/32/0/1/manual_time                -0.0514         -0.0440            40            38            40            38
ParquetWrite/timestamps_void_output/32/0/1/1/2/manual_time                      -0.0087         -0.0017           171           170           170           169
ParquetWrite/timestamps_void_output/32/1000/1/1/2/manual_time                   -0.0396         -0.0346            50            48            49            48
ParquetWrite/timestamps_void_output/32/0/32/1/2/manual_time                     -0.0497         -0.0473            42            40            42            40
ParquetWrite/timestamps_void_output/32/1000/32/1/2/manual_time                  -0.0762         -0.0536            33            30            32            30
ParquetWrite/timestamps_void_output/32/0/1/0/2/manual_time                      -0.0655         -0.0740            25            23            25            23
ParquetWrite/timestamps_void_output/32/1000/1/0/2/manual_time                   -0.0803         -0.0542            38            35            36            34
ParquetWrite/timestamps_void_output/32/0/32/0/2/manual_time                     -0.0421         -0.0629            30            29            30            28
ParquetWrite/timestamps_void_output/32/1000/32/0/2/manual_time                  -0.0655         -0.0591            31            29            31            29
ParquetWrite/string_file_output/23/0/1/1/0/manual_time                          -0.0521         -0.0492          1631          1546           537           510
ParquetWrite/string_file_output/23/1000/1/1/0/manual_time                       -0.0056         -0.0047           193           191            95            95
ParquetWrite/string_file_output/23/0/32/1/0/manual_time                         -0.0015         -0.0176            87            87            50            49
ParquetWrite/string_file_output/23/1000/32/1/0/manual_time                      -0.0374         -0.0472            55            53            38            36
ParquetWrite/string_file_output/23/0/1/0/0/manual_time                          -0.0659         +0.0329          1584          1480           406           420
ParquetWrite/string_file_output/23/1000/1/0/0/manual_time                       -0.0614         -0.0179           194           182            89            88
ParquetWrite/string_file_output/23/0/32/0/0/manual_time                         +0.0103         -0.0483            86            86            47            45
ParquetWrite/string_file_output/23/1000/32/0/0/manual_time                      -0.0477         -0.0213            55            52            36            35
ParquetWrite/string_buffer_output/23/0/1/1/1/manual_time                        -0.0040         -0.0033           881           878           880           878
ParquetWrite/string_buffer_output/23/1000/1/1/1/manual_time                     -0.0206         -0.0199           110           108           109           107
ParquetWrite/string_buffer_output/23/0/32/1/1/manual_time                       -0.1172         -0.1106            66            58            65            58
ParquetWrite/string_buffer_output/23/1000/32/1/1/manual_time                    -0.0503         -0.0598            40            38            39            37
ParquetWrite/string_buffer_output/23/0/1/0/1/manual_time                        -0.0027         -0.0028           790           788           790           788
ParquetWrite/string_buffer_output/23/1000/1/0/1/manual_time                     +0.0066         +0.0037            97            98            96            96
ParquetWrite/string_buffer_output/23/0/32/0/1/manual_time                       -0.1049         -0.1125            58            52            57            51
ParquetWrite/string_buffer_output/23/1000/32/0/1/manual_time                    -0.0598         -0.0603            39            36            38            36
ParquetWrite/string_void_output/23/0/1/1/2/manual_time                          -0.0043         -0.0063           155           155           155           154
ParquetWrite/string_void_output/23/1000/1/1/2/manual_time                       -0.0217         -0.0281            51            50            51            49
ParquetWrite/string_void_output/23/0/32/1/2/manual_time                         -0.0336         -0.0358            33            31            32            31
ParquetWrite/string_void_output/23/1000/32/1/2/manual_time                      -0.0396         -0.0438            29            28            29            28
ParquetWrite/string_void_output/23/0/1/0/2/manual_time                          -0.0500         -0.0431            37            35            36            35
ParquetWrite/string_void_output/23/1000/1/0/2/manual_time                       -0.0391         -0.0325            44            42            43            42
ParquetWrite/string_void_output/23/0/32/0/2/manual_time                         -0.0363         -0.0450            27            26            26            25
ParquetWrite/string_void_output/23/1000/32/0/2/manual_time                      -0.0278         -0.0520            28            27            27            26
ParquetWrite/list_file_output/24/0/1/1/0/manual_time                            +0.0191         +0.0030          1631          1662           888           890
ParquetWrite/list_file_output/24/1000/1/1/0/manual_time                         -0.0277         +0.0015          3263          3173          2715          2719
ParquetWrite/list_file_output/24/0/32/1/0/manual_time                           +0.0290         +0.0006           541           556           487           488
ParquetWrite/list_file_output/24/1000/32/1/0/manual_time                        +0.0597         +0.0168           517           547           485           493
ParquetWrite/list_file_output/24/0/1/0/0/manual_time                            +0.0194         -0.0119          1264          1289           522           516
ParquetWrite/list_file_output/24/1000/1/0/0/manual_time                         -0.0043         -0.0105          1273          1268           531           526
ParquetWrite/list_file_output/24/0/32/0/0/manual_time                           +0.0379         -0.0055          1150          1193           466           463
ParquetWrite/list_file_output/24/1000/32/0/0/manual_time                        -0.0464         +0.0122          1340          1278           491           497
ParquetWrite/list_buffer_output/24/0/1/1/1/manual_time                          +0.0126         +0.0126          1276          1292          1276          1292
ParquetWrite/list_buffer_output/24/1000/1/1/1/manual_time                       -0.0019         -0.0019          2914          2908          2914          2908
ParquetWrite/list_buffer_output/24/0/32/1/1/manual_time                         -0.0055         -0.0055           503           500           503           500
ParquetWrite/list_buffer_output/24/1000/32/1/1/manual_time                      -0.0078         -0.0078           506           502           506           502
ParquetWrite/list_buffer_output/24/0/1/0/1/manual_time                          -0.0459         -0.0459           916           874           916           874
ParquetWrite/list_buffer_output/24/1000/1/0/1/manual_time                       -0.0048         -0.0048           846           842           846           842
ParquetWrite/list_buffer_output/24/0/32/0/1/manual_time                         -0.0116         -0.0116           857           847           857           847
ParquetWrite/list_buffer_output/24/1000/32/0/1/manual_time                      +0.0632         +0.0632           716           761           716           761
ParquetWrite/list_void_output/24/0/1/1/2/manual_time                            +0.0061         +0.0061           594           598           594           598
ParquetWrite/list_void_output/24/1000/1/1/2/manual_time                         -0.0000         -0.0000          2505          2505          2505          2505
ParquetWrite/list_void_output/24/0/32/1/2/manual_time                           -0.0001         -0.0002           444           444           444           444
ParquetWrite/list_void_output/24/1000/32/1/2/manual_time                        -0.0026         -0.0026           451           449           451           449
ParquetWrite/list_void_output/24/0/1/0/2/manual_time                            +0.0006         +0.0006           186           186           186           186
ParquetWrite/list_void_output/24/1000/1/0/2/manual_time                         +0.0021         +0.0021           204           205           204           205
ParquetWrite/list_void_output/24/0/32/0/2/manual_time                           +0.0001         +0.0001           159           159           159           159
ParquetWrite/list_void_output/24/1000/32/0/2/manual_time                        +0.0014         +0.0014           160           160           160           160
ParquetWrite/writer_options/0/0/0/manual_time                                   -0.0240         -0.0061           267           260           142           141
ParquetWrite/writer_options/2/0/0/manual_time                                   -0.0009         +0.0041           286           285           180           181
ParquetWrite/writer_options/0/1/0/manual_time                                   +0.0302         +0.0071           251           259           145           146
ParquetWrite/writer_options/2/1/0/manual_time                                   -0.0241         +0.0066           290           283           184           186
ParquetWrite/writer_options/0/2/0/manual_time                                   +0.0110         -0.0111           260           263           147           145
ParquetWrite/writer_options/2/2/0/manual_time                                   +0.0062         +0.0021           287           289           183           183
ParquetWrite/writer_options/0/0/1/manual_time                                   +0.0262         -0.0127           253           260           144           142
ParquetWrite/writer_options/2/0/1/manual_time                                   -0.0331         -0.0009           277           268           180           180
ParquetWrite/writer_options/0/1/1/manual_time                                   -0.0644         -0.0115           282           264           148           146
ParquetWrite/writer_options/2/1/1/manual_time                                   +0.0067         +0.0172           270           272           182           185
ParquetWrite/writer_options/0/2/1/manual_time                                   +0.0595         -0.0116           259           274           148           146
ParquetWrite/writer_options/2/2/1/manual_time                                   +0.0066         +0.0062           280           282           183           184
(cudf_dev) dmakkar@devavret-X99-E:/Users/dmakkar/Developer/Porting/cudf/cpp/build$ ./googlebenchmark/googlebenchmark/tools/compare.py benchmarks oldread newread 
Comparing oldread to newread
Benchmark                                                                        Time             CPU      Time Old      Time New       CPU Old       CPU New
-------------------------------------------------------------------------------------------------------------------------------------------------------------
ParquetRead/integral_file_input/28/0/1/1/0/manual_time                        -0.0021         -0.0022          1123          1121          1123          1121
ParquetRead/integral_file_input/28/1000/1/1/0/manual_time                     -0.0009         +0.0015           285           285           284           284
ParquetRead/integral_file_input/28/0/32/1/0/manual_time                       +0.0136         +0.0003           257           260           257           257
ParquetRead/integral_file_input/28/1000/32/1/0/manual_time                    -0.0029         +0.0111            50            50            49            50
ParquetRead/integral_file_input/28/0/1/0/0/manual_time                        +0.0090         +0.0025           119           120           119           119
ParquetRead/integral_file_input/28/1000/1/0/0/manual_time                     +0.0074         +0.0052            60            60            60            60
ParquetRead/integral_file_input/28/0/32/0/0/manual_time                       +0.0422         +0.0041            29            30            29            29
ParquetRead/integral_file_input/28/1000/32/0/0/manual_time                    -0.0085         -0.0133            27            27            27            26
ParquetRead/integral_buffer_input/28/0/1/1/1/manual_time                      -0.0028         -0.0042          1105          1102          1105          1100
ParquetRead/integral_buffer_input/28/1000/1/1/1/manual_time                   +0.0036         +0.0035           274           275           274           275
ParquetRead/integral_buffer_input/28/0/32/1/1/manual_time                     +0.0078         -0.0009           256           258           256           256
ParquetRead/integral_buffer_input/28/1000/32/1/1/manual_time                  -0.0052         -0.0147            49            49            49            49
ParquetRead/integral_buffer_input/28/0/1/0/1/manual_time                      -0.0056         -0.0102            95            95            95            94
ParquetRead/integral_buffer_input/28/1000/1/0/1/manual_time                   +0.0131         +0.0027            53            54            53            53
ParquetRead/integral_buffer_input/28/0/32/0/1/manual_time                     +0.0104         -0.0003            28            28            28            28
ParquetRead/integral_buffer_input/28/1000/32/0/1/manual_time                  +0.0015         -0.0051            27            27            26            26
ParquetRead/floats_file_input/30/0/1/1/0/manual_time                          +0.0083         +0.0082            97            98            97            98
ParquetRead/floats_file_input/30/1000/1/1/0/manual_time                       +0.0028         +0.0028           199           200           199           200
ParquetRead/floats_file_input/30/0/32/1/0/manual_time                         -0.0687         -0.0068            20            18            18            18
ParquetRead/floats_file_input/30/1000/32/1/0/manual_time                      -0.0112         +0.0011            16            16            15            15
ParquetRead/floats_file_input/30/0/1/0/0/manual_time                          +0.0157         +0.0158            99           101            99           101
ParquetRead/floats_file_input/30/1000/1/0/0/manual_time                       +0.0076         -0.0025            39            40            39            39
ParquetRead/floats_file_input/30/0/32/0/0/manual_time                         +0.0852         +0.0168            18            20            18            18
ParquetRead/floats_file_input/30/1000/32/0/0/manual_time                      +0.0097         +0.0136            16            16            15            15
ParquetRead/floats_buffer_input/30/0/1/1/1/manual_time                        -0.0018         -0.0018            78            78            78            78
ParquetRead/floats_buffer_input/30/1000/1/1/1/manual_time                     -0.0012         -0.0012           193           193           194           193
ParquetRead/floats_buffer_input/30/0/32/1/1/manual_time                       +0.0385         +0.0068            18            18            17            17
ParquetRead/floats_buffer_input/30/1000/32/1/1/manual_time                    -0.0071         +0.0110            15            15            15            15
ParquetRead/floats_buffer_input/30/0/1/0/1/manual_time                        +0.0104         +0.0103            78            78            78            78
ParquetRead/floats_buffer_input/30/1000/1/0/1/manual_time                     +0.0010         -0.0036            33            33            33            33
ParquetRead/floats_buffer_input/30/0/32/0/1/manual_time                       +0.0781         +0.0055            18            19            17            17
ParquetRead/floats_buffer_input/30/1000/32/0/1/manual_time                    +0.0062         -0.0002            15            15            15            15
ParquetRead/timestamps_file_input/32/0/1/1/0/manual_time                      +0.0015         +0.0014          1126          1128          1126          1128
ParquetRead/timestamps_file_input/32/1000/1/1/0/manual_time                   +0.0024         +0.0004           161           161           161           161
ParquetRead/timestamps_file_input/32/0/32/1/0/manual_time                     -0.0014         -0.0014           156           156           156           156
ParquetRead/timestamps_file_input/32/1000/32/1/0/manual_time                  +0.0136         +0.0183            26            27            26            27
ParquetRead/timestamps_file_input/32/0/1/0/0/manual_time                      -0.0030         -0.0030            97            97            97            97
ParquetRead/timestamps_file_input/32/1000/1/0/0/manual_time                   +0.0047         +0.0058            33            33            32            33
ParquetRead/timestamps_file_input/32/0/32/0/0/manual_time                     -0.0148         +0.0025            17            17            16            16
ParquetRead/timestamps_file_input/32/1000/32/0/0/manual_time                  +0.0131         +0.0036            13            14            13            13
ParquetRead/timestamps_buffer_input/32/0/1/1/1/manual_time                    +0.0066         +0.0066          1104          1112          1104          1112
ParquetRead/timestamps_buffer_input/32/1000/1/1/1/manual_time                 +0.0002         +0.0003           156           156           156           156
ParquetRead/timestamps_buffer_input/32/0/32/1/1/manual_time                   -0.0004         -0.0004           154           154           154           154
ParquetRead/timestamps_buffer_input/32/1000/32/1/1/manual_time                +0.0223         +0.0051            26            27            26            26
ParquetRead/timestamps_buffer_input/32/0/1/0/1/manual_time                    +0.0070         +0.0070            76            76            76            76
ParquetRead/timestamps_buffer_input/32/1000/1/0/1/manual_time                 +0.0039         +0.0097            29            29            28            28
ParquetRead/timestamps_buffer_input/32/0/32/0/1/manual_time                   -0.0552         -0.0110            16            15            15            15
ParquetRead/timestamps_buffer_input/32/1000/32/0/1/manual_time                +0.0074         +0.0030            13            14            13            13
ParquetRead/string_file_input/23/0/1/1/0/manual_time                          -0.0028         -0.0001          1363          1359          1359          1359
ParquetRead/string_file_input/23/1000/1/1/0/manual_time                       -0.0122         +0.0002           100            99            99            99
ParquetRead/string_file_input/23/0/32/1/0/manual_time                         -0.0057         -0.0057            82            82            82            82
ParquetRead/string_file_input/23/1000/32/1/0/manual_time                      -0.0095         -0.0110            45            45            45            45
ParquetRead/string_file_input/23/0/1/0/0/manual_time                          -0.0324         -0.0264           164           159           163           159
ParquetRead/string_file_input/23/1000/1/0/0/manual_time                       +0.0030         -0.0032            44            44            44            44
ParquetRead/string_file_input/23/0/32/0/0/manual_time                         -0.0012         -0.0042            34            34            35            34
ParquetRead/string_file_input/23/1000/32/0/0/manual_time                      +0.0207         +0.0154            31            31            31            31
ParquetRead/string_buffer_input/23/0/1/1/1/manual_time                        -0.0005         -0.0005          1329          1328          1329          1328
ParquetRead/string_buffer_input/23/1000/1/1/1/manual_time                     +0.0136         +0.0151            98            99            96            98
ParquetRead/string_buffer_input/23/0/32/1/1/manual_time                       +0.0090         +0.0091            81            81            81            81
ParquetRead/string_buffer_input/23/1000/32/1/1/manual_time                    -0.0027         -0.0029            44            44            45            44
ParquetRead/string_buffer_input/23/0/1/0/1/manual_time                        -0.0061         -0.0038           129           128           128           128
ParquetRead/string_buffer_input/23/1000/1/0/1/manual_time                     +0.0094         +0.0026            41            41            41            41
ParquetRead/string_buffer_input/23/0/32/0/1/manual_time                       -0.0086         -0.0099            33            33            33            33
ParquetRead/string_buffer_input/23/1000/32/0/1/manual_time                    +0.0009         -0.0002            30            31            31            30
ParquetRead/list_file_input/24/0/1/1/0/manual_time                            +0.0000         +0.0000         40216         40218         40217         40219
ParquetRead/list_file_input/24/1000/1/1/0/manual_time                         -0.0001         -0.0001         39716         39711         39716         39712
ParquetRead/list_file_input/24/0/32/1/0/manual_time                           +0.0000         +0.0000          5787          5787          5787          5787
ParquetRead/list_file_input/24/1000/32/1/0/manual_time                        +0.0006         +0.0006          5423          5426          5423          5426
ParquetRead/list_file_input/24/0/1/0/0/manual_time                            +0.0054         +0.0055           677           681           677           681
ParquetRead/list_file_input/24/1000/1/0/0/manual_time                         +0.0040         +0.0039           716           719           716           719
ParquetRead/list_file_input/24/0/32/0/0/manual_time                           +0.0021         +0.0019           593           595           593           595
ParquetRead/list_file_input/24/1000/32/0/0/manual_time                        +0.0101         +0.0101           592           598           592           598
ParquetRead/list_buffer_input/24/0/1/1/1/manual_time                          +0.0000         +0.0000         40197         40197         40197         40198
ParquetRead/list_buffer_input/24/1000/1/1/1/manual_time                       +0.0001         +0.0001         39695         39701         39696         39702
ParquetRead/list_buffer_input/24/0/32/1/1/manual_time                         -0.0011         -0.0011          5795          5788          5795          5788
ParquetRead/list_buffer_input/24/1000/32/1/1/manual_time                      +0.0001         +0.0001          5417          5417          5417          5417
ParquetRead/list_buffer_input/24/0/1/0/1/manual_time                          +0.0047         +0.0049           650           653           650           653
ParquetRead/list_buffer_input/24/1000/1/0/1/manual_time                       +0.0011         +0.0011           689           690           689           690
ParquetRead/list_buffer_input/24/0/32/0/1/manual_time                         +0.0013         +0.0012           568           569           568           569
ParquetRead/list_buffer_input/24/1000/32/0/1/manual_time                      -0.0004         -0.0004           570           570           570           570
ParquetRead/column_selection/0/0/1/1/0/manual_time                            -0.0001         -0.0001           334           334           334           334
ParquetRead/column_selection/1/0/1/1/0/manual_time                            +0.0043         +0.0043           322           324           322           324
ParquetRead/column_selection/2/0/1/1/0/manual_time                            +0.0015         +0.0015           322           322           322           322
ParquetRead/column_selection/3/0/1/1/0/manual_time                            -0.0002         -0.0002           324           324           324           324
ParquetRead/row_selection/0/5/1/1/0/manual_time                               -0.0048         -0.0019           336           335           334           334
ParquetRead/row_selection/0/2/1/1/0/manual_time                               +0.0014         +0.0014           334           335           334           335
ParquetRead/row_selection/0/5/4/1/0/manual_time                               +0.0050         +0.0036          1269          1275          1269          1273
ParquetRead/row_selection/0/2/4/1/0/manual_time                               +0.0009         +0.0009          1278          1279          1278          1279
ParquetRead/misc_options/0/2/1/1/0/manual_time                                -0.0002         -0.0002           334           334           334           334
ParquetRead/misc_options/0/2/1/0/0/manual_time                                -0.0015         -0.0016           337           336           337           336
ParquetRead/misc_options/0/2/1/3/0/manual_time                                +0.0017         +0.0000           334           335           334           334
ParquetRead/misc_options/0/2/1/2/0/manual_time                                -0.0054         -0.0015           338           336           336           336
ParquetRead/misc_options/0/2/1/1/16/manual_time                               +0.0102         +0.0045           334           338           334           336
ParquetRead/misc_options/0/2/1/0/16/manual_time                               +0.0001         +0.0000           337           337           337           337
ParquetRead/misc_options/0/2/1/3/16/manual_time                               -0.0092         -0.0011           338           334           335           334
ParquetRead/misc_options/0/2/1/2/16/manual_time                               +0.0000         -0.0000           337           337           337           337
```


</p>
</details>

File: cpp/src/io/parquet/writer_impl.cu (hunk 1 above)
Comment (devavret) [reply cpp/src/io/parquet/writer_impl.cu:692]: #7873

**Review by hyperbolic2346 — APPROVED** [review]

**Comment by devavret** [comment]
@gpucibot merge

**Pull request merged**